packages/hoppscotch-backend/src/user-collection/user-collection.resolver.ts

```diff
@@ -142,7 +143,13 @@ export class UserCollectionResolver {
     );
 
     if (E.isLeft(userCollection)) throwErr(userCollection.left);
-    return userCollection.right;
+    return <UserCollection>{
+      ...userCollection.right,
+      userID: userCollection.right.userUid,
+      data: !userCollection.right.data
+        ? null
+        : JSON.stringify(userCollection.right.data),
+    };
```

Comment:
We can move type conversion to service file (inside `cast` function)

Reply:
Same reasoning as why I did this for `TeamCollection` module.